opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java

```diff
@@ -164,4 +171,93 @@ public void testEventOnCreationAndDeletion() throws Exception {
         // delete the one with polling packages
         sendRequest(DELETE, "/monitoringLocations/location2", 204);
     }
+
+    /**
+     * Add more locations than the previous limit of 10
+     * Store them locally as we add them
+     * Then fetch the list, sort and compare the lists
+     */
+    @Test
+    @Transactional
+    public void testListLimitAndSort() throws Exception {
+        final int LOCATION_COUNT= ThreadLocalRandom.current().nextInt(11, 20); // original default limit was 10
+        ArrayList<Location> localLocations = new ArrayList<>(LOCATION_COUNT);
+
+        // Add default location but do not post it as it is already there
+        Location defaultLoc = new Location();
+        defaultLoc.locationName = "Default";
+        defaultLoc.monitoringArea = "localhost";
+        localLocations.add(defaultLoc);
+
+        // Add locations in reverse order so they can be sorted
+        for (int i = LOCATION_COUNT - 1; i >= 0 ; i--) {
+            Location loc = new Location();
+            loc.locationName = "location-"+i;
+            loc.monitoringArea = "rand-area-"+i;
+            localLocations.add(loc);
+
+            String location = "<location location-name=\"" + loc.locationName + "\" monitoring-area=\""+loc.monitoringArea+"\"/>";
+            sendPost("/monitoringLocations", location, 201, null);
+        }
+
+        // Fetch count and check it against local count
+        String remoteCount = sendRequest(GET, "/monitoringLocations/count", Collections.emptyMap(), 200);
+        LOG.info("testListLimitAndSort: remoteCount="+remoteCount+" localCount="+localLocations.size());
+        assertTrue (localLocations.size() == Integer.parseInt(remoteCount));
```

Comment:
Avoid `assertTrue` when possible - if it assertion does fail, the error message it gives isn't helpful. This is better written as:
```
assertEquals(localLocations.size(), Integer.parseInt(remoteCount))
```
or
```
assertThat(Integer.parseInt(remoteCount), equalTo(localLocations.size()))
```